riscv/debug_module.cc

```diff
@@ -396,23 +396,25 @@ bool debug_module_t::dmi_write(unsigned address, uint32_t value)
   D(fprintf(stderr, "dmi_write(0x%x, 0x%x)\n", address, value));
   if (address >= DMI_DATA0 && address < DMI_DATA0 + abstractcs.datacount) {
     unsigned i = address - DMI_DATA0;
-    write32(dmdata, address - DMI_DATA0, value);
+    if (!abstractcs.busy)
```

Comment:
I don't think the HW prevents this sort of access to DATA while busy. Is it necessary for correct behavior? It just flags BUSY error if you do it, but doesn't actually block the write (I think).